os/mm/mm_heap/mm_getheap.c

```diff
@@ -51,9 +51,93 @@ extern struct mm_heap_s g_mmheap[CONFIG_MM_NHEAPS];
  * Pre-processor Definitions
  ****************************************************************************/
 
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+typedef struct app_heap_s {
+	struct mm_heap_s *heap;
+	struct app_heap_s *next;
+	struct app_heap_s *prev;
```

Comment:
Why not use internal queue structure for maintaining linked list ?

Reply:
I cannot use the queue implementation because i need the functionality of a list here. 
I need to be able to search the elements in the list and selectively remove an element from the middle of the list. 
This is not possible with a queue.

Reply:
But it has support to remove any element from the queue either last, first or middle.